Common/include/option_structure.hpp

```diff
@@ -912,51 +912,52 @@ static const MapType<std::string, ENUM_LIMITER> Limiter_Map = {
 enum class TURB_MODEL {
   NONE,      /*!< \brief No turbulence model. */
   SA,        /*!< \brief Kind of Turbulent model (Spalart-Allmaras). */
-  SA_NEG,    /*!< \brief Kind of Turbulent model (Spalart-Allmaras). */
-  SA_E,      /*!< \brief Kind of Turbulent model (Spalart-Allmaras Edwards). */
-  SA_COMP,   /*!< \brief Kind of Turbulent model (Spalart-Allmaras Compressibility Correction). */
-  SA_E_COMP, /*!< \brief Kind of Turbulent model (Spalart-Allmaras Edwards with Compressibility Correction). */
   SST,       /*!< \brief Kind of Turbulence model (Menter SST). */
-  SST_SUST   /*!< \brief Kind of Turbulence model (Menter SST with sustaining terms for free-stream preservation). */
 };
 static const MapType<std::string, TURB_MODEL> Turb_Model_Map = {
   MakePair("NONE", TURB_MODEL::NONE)
   MakePair("SA", TURB_MODEL::SA)
-  MakePair("SA_NEG", TURB_MODEL::SA_NEG)
-  MakePair("SA_E", TURB_MODEL::SA_E)
-  MakePair("SA_COMP", TURB_MODEL::SA_COMP)
-  MakePair("SA_E_COMP", TURB_MODEL::SA_E_COMP)
   MakePair("SST", TURB_MODEL::SST)
-  MakePair("SST_SUST", TURB_MODEL::SST_SUST)
 };
 
-/*!
- * \brief Families of turbulence models
- */
-enum class TURB_FAMILY {
-  NONE,   /*!< \brief No turbulence model. */
-  SA,     /*!< \brief Spalart-Allmaras variants. */
-  KW,     /*!< \brief k-w models. */
+enum class SA_OPTIONS {
+  NONE,     /*!< \brief Stadard Spalart-Allmaras (SA_ Model. */
+  NEG,      /*!< \brief Negative SA model. */
+  E,        /*!< \brief SA with Edwards modificiation. */
+  COMP,     /*!< \brief SA model with no tau_ij in momentum and energy eqns. */
+  E_COMP,   /*!< \brief SA with Edwards modification and compressibility correction. */
+};
+static const MapType<std::string, SA_OPTIONS> SA_Options_Map = {
+  MakePair("NONE", SA_OPTION::NONE)
+  MakePair("NEG", SA_OPTION::NEG)
+  MakePair("E", SA_OPTION::E)
+  MakePair("COMP", SA_OPTION::COMP)
+  MakePair("E_COMP", SA_OPTION::E_COMP)
+};
+
+/*!
+ * \brief Variants and corrections of SST k-w models
+ */
+enum class SST_OPTIONS {
```

Comment:
This causes some problems since (at least as Im aware) the SST options can be used in almost set of permutations.  This is slightly different to SA.

If we want to allow for these types of modifications, In config.cpp, I propose to something similiar to #1413, but the opposite.  i.e We take in a list of the options and create/name the desired model from there.  This would make a large decision tree however.

Reply:
We don't want to unfold these options into turbulence models. What we can do is read an enum list from the config, as you have started defining, then we convert that list into a small struct with sst boolean options and we validate that they are compatible.
```
Version: enum - 1994 or 2003
Modified: bool - m
Production: enum - V or KL or UQ
Sustaining terms: bool - sust
Curvature correction: bool - RC
```
Then we use that struct in the code instead of deciding what to do based on kind turb model.